espressif/vscode-esp-idf-extension
# prefer computed python venv path

## Description

Remove `idf.pythonBinPath`, `idf.adapterTargetName`, `idf.customAdapterTargetName`, `idf.customExtraPaths`, `idf.selectedDfuDevicePath` and `idf.listDfuDevices` in favor of calculating them from IDF_PATH and IDF_TOOLS_PATH.

Doctor command now computes ESP-IDF tools to be added in PATH and the used venv from IDF_PATH and IDF_TOOLS_PATH.

This should improve maintenance of extension, reduce setup issues and reduce regular docker file updates.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Steps to test this pull request

This PR changes impact all the extension. It is necessary a full extension review for this PR to review avoid breaking existing features.

A basic test flow should include: Configure ESP-IDF extension, create project, build, flash monitor and debug as described in documentation.

Other features that requires ESP-IDF Tools or Virtual environment Python should also be review like: App Tracing, IDF Size UI and others. Doctor command now

- Expected behaviour: All extension features work the same as before but used settings are less than before. 

- Expected output:

## How has this been tested?

Ideally user should test this PR with all main features by following commands. A basic test flow should include: Configure ESP-IDF extension, create project, build, flash monitor and debug as described in documentation.

**Test Configuration**:
* ESP-IDF Version: 5.2
* OS (Windows,Linux and macOS): MacOS and Windows

## Checklist
